Library/Homebrew/formula.rb

```diff
@@ -1008,7 +1008,7 @@ def caveats
   # rarely, you don't want your library symlinked into the main prefix
   # see gettext.rb for an example
   def keg_only?
-    keg_only_reason && keg_only_reason.valid?
+    !!keg_only_reason && keg_only_reason.valid?
```

Comment:
Alternatively, we could make `keg_only` without a reason a run-time failure with a "please supply a reason, fool" message, though being able to just add `keg_only` for testing purposes is kind of a good thing. Perhaps lack of a reason is best left to audit to complain about.